site/oauth2.md

```diff
@@ -0,0 +1,464 @@
+<!--
+Copyright (c) 2007-2022 VMware, Inc. or its affiliates.
+
+All rights reserved. This program and the accompanying materials
+are made available under the terms of the under the Apache License,
+Version 2.0 (the "License”); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# OAuth 2.0 Authentication Backend
+
+## <a id="overview" class="anchor" href="#overview">Overview</a>
+
+This [RabbitMQ authentication/authorisation backend](https://www.rabbitmq.com/access-control.html) plugin lets applications (clients) and users authenticate and authorize using JWT-encoded OAuth 2.0 access tokens.
+
+This guide covers
+
+ * [How it works](#how-it-works)
+ * [Usage](#usage)
+    * [Variables Configurable in rabbitmq.conf](#variables-configurable)
+    * [Resource Server ID and Scope Prefixes](#resource-server-id)
+    * [Token validation](#token-validation)
+    * [Scope-to-Permission Translation](#scope-transalation)
+    * [Using a different token field for the Scope](#use-different-token-field)
+    * [Using Tokens with Clients](#use-tokens-with-clients)
+    * [Scope and Tags](#scope-and-tags)
+    * [Token Expiration and Refresh](#token-expiration)
+    * [Rich Authorization Request](#rich-authorization-request)
+ * [Examples](#examples)
+
+
+## <a id="how-it-works" class="anchor" href="#how-it-works">How it works</a>
+
+### <a id="authorization-workflow" class="anchor" href="#authorization-workflow">Authorization Workflow</a>
+
+This plugin does not communicate with any OAuth 2.0 provider. It decodes an access token provided by the client and authorises a user based on the data stored in the token.
+
+The token can be any [JWT token](https://jwt.io/introduction/) which contains the `scope` and `aud` fields. The way the token was issued (such as what grant type was used) is outside of the scope of this plugin.
```

Comment:
The token can be any [JWT token](https://jwt.io/introduction/) which contains the `scope` and `aud` fields. The way the token is issued (such as what grant type is used) is outside of the scope of this plugin.